packages/react-resizable-panels/package.json

```diff
@@ -9,11 +9,33 @@
     "url": "git+https://github.com/bvaughn/react-resizable-panels.git"
   },
   "source": "src/index.ts",
-  "main": "dist/react-resizable-panels.js",
-  "module": "dist/react-resizable-panels.module.js",
-  "types": "dist/react-resizable-panels.d.ts",
+  "main": "dist/react-resizable-panels.cjs.js",
+  "module": "dist/react-resizable-panels.esm.js",
+  "exports": {
+    ".": {
+      "types": {
+        "import": "./dist/react-resizable-panels.cjs.mjs",
+        "default": "./dist/react-resizable-panels.cjs.js"
+      },
+      "development": {
+        "module": "./dist/react-resizable-panels.development.esm.js",
+        "import": "./dist/react-resizable-panels.development.cjs.mjs",
+        "default": "./dist/react-resizable-panels.development.cjs.js"
+      },
+      "module": "./dist/react-resizable-panels.esm.js",
+      "import": "./dist/react-resizable-panels.cjs.mjs",
+      "default": "./dist/react-resizable-panels.cjs.js"
+    },
+    "./package.json": "./package.json"
+  },
+  "imports": {
+    "#is-development": {
+      "development": "./src/env-conditions/development.ts",
+      "default": "./src/env-conditions/production.ts"
+    }
+  },
```

Comment:
You can freely reorganize this, only keys (aka conditions) are relevant (and the fact that they are pointing to files in `src`). I'm just experimenting with how to best organize those files - their implementation can also be just anything but we use simple named exports with boolean values in them here. The recommendation is for all files used for a certain `imports` to have the same module interface (same types) - otherwise it could lead to weird problems, remember that they should be "swappable".

Those are only used in **development** (when using Preconstruct, since those are valid things in node etc and u could leave imports to them as-is). `preconstruct build` inlines/bundles those files so all of that should still work with older runtime/bundlers/etc